doc/edgehub-api.md

```diff
@@ -0,0 +1,434 @@
+
+### Receiving Cloud-to-Device messages
+Just like other message types, C2D messages cannot be received until a device subscribes to them. C2D messages cannot be sent to modules, only to devices. When a device is ready to receive C2D messages, it subscribes to the following topic:
+```
+devices/{device_id}/messages/devicebound/#
+```
+When IoT Hub sends a message, the it will be published on the topic similar to the following:
+```
+devices/dev_c/messages/devicebound/propertyName=propertyValue&%24.to=%2Fdevices%2Fdev_c%2Fmessages%2FdeviceBound
+```
+In this example the device name is 'dev_c'. The topic encodes a user property called 'propertyName', which has the value 'propertyValue'. Also, there is a system propert added to the topic ('$.to').
+
+The response to a C2D message at application level can be 'complete', 'reject' and 'abandon', however the MQTT protocol is not able to express all of these. When a C2D message is received by a device and the device acknowledges the MQTT packet that delivered the C2D message, Edge Hub takes the acknowledgement as 'complete' and forwards this back to IoT Hub. If the receiving client does not acknowledges the MQTT packet, IoT Hub is going to be timeout after while and it takes it as 'abandon'.
+
+### Receiving Direct-Method calls
+
+Similarly to C2D messages and twins, Direct-Method calls need to be subscribed to. IoT Hub and Edge Hub publishes direct method calls on a topic:
+```
+$iothub/methods/POST/{method name}/?$rid={request id}
+``` 
+According to that publication, a client that wants to receive direct method calls needs to subsribe to:
+```
+$iothub/methods/POST/#
+``` 
+When IoT Hub sends a direct method call, Edge Hub relays that call and publishes an MQTT message like the following:
+```
+$iothub/methods/POST/some_method/?$rid=505e09bb-0076-4b9f-b4a3-529430f1593f
+```
+This call has the method name "some_method" and a guid as request id. This request id must be sent back with the response payload.
+The payload of the direct method call can be an arbitrary json document:
+```
+{
+    "some_property" : "some_value",
+    "some_other_property" : 123
+}
+```
+When the client processed the direct method call, it can send back a payload and also a status code as a result of a call. This status code can be any number that the caller application can interpret. Also, the payload can be any json document. An example response to the direct method call earlier can be published to the following topic:
+```
+$iothub/methods/res/200/?$rid=505e09bb-0076-4b9f-b4a3-529430f1593f
+```
+There is a configurable time limit the device/module needs to answer, which is 30 seconds by default, but can be set from 5 to 300 seconds.
+## Edge Hub 1.2 MQTT Endpoint Extensions
```

Comment:
Nit: Let's not call this "1.2". It is current MQTT Endpoint Extensions. 
We should also mention explicitly and in bold that this is experimental and subject to change.